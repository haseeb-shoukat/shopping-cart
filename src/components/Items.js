import Product from "./Product";
import { v4 as uuidv4 } from "uuid";

const Items = [
  new Product(uuidv4(), "Matt Black Cap, Unbranded, Unisex", 19.5, "cap"),
  new Product(uuidv4(), "Nike Orange-White Jacket", 155, "jacket"),
  new Product(uuidv4(), "Plain White T-Shirt for Men", 60, "tshirt"),
  new Product(
    uuidv4(),
    "Red Butterfly Dress + Black Belt for Women",
    245,
    "dress"
  ),
  new Product(uuidv4(), "Rayban 2020 Limited Edition Shades", 250, "glasses"),
  new Product(
    uuidv4(),
    "Stylish Denim Jeans for Women, Tight Fit",
    30,
    "jeans"
  ),
  new Product(uuidv4(), "Pink Polo Shirt, Slim Waist for Women", 20, "polo"),
  new Product(uuidv4(), "Plain Black Shirt, Fitted, Men", 40, "shirt"),
  new Product(uuidv4(), "Stylish High Heels, Full Black", 250, "shoes"),
  new Product(uuidv4(), "Class Release, Two Piece Black Suit", 400, "suit"),
];

export default Items;
