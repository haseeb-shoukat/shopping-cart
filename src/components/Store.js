import { useEffect } from "react";

const Store = ({ products }) => {
  useEffect(() => {
    document.querySelector(".active").classList.remove("active");
    document.querySelector("#products").classList.add("active");
  }, []);

  return (
    <div className="products">
      {products.map((product) => {
        return (
          <div key={product.key} className="product-card">
            <img
              className="card-img"
              src={`./images/clothing/${product.img}.png`}
              alt=""
            />
            <div className="card-body">
              <div className="card-info">
                <div className="product-name">{product.name}</div>
                <div className="product-price">${product.price.toFixed(2)}</div>
              </div>
              <button className="cart-btn">Add To Cart</button>
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default Store;
