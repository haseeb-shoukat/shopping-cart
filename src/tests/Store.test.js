import React from "react";
import { getAllByRole, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import Store from "../components/Store";
import Items from "../components/Items";

describe("Store component", () => {
  const store = global.document.createElement("div");
  store.id = "products";
  global.document.body.appendChild(store);

  const updateProducts = jest.fn();

  it("Renders successfully", () => {
    const { container } = render(
      <Store products={Items} updateProducts={updateProducts} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Calls updateTotal/product.increment when Add to Cart button is pressed", () => {
    const fakeItem = {
      key: 1,
      image: "",
      name: "",
      price: 60,
      totalPrice: 60,
      increment: jest.fn(),
    };
    render(<Store products={[fakeItem]} updateProducts={updateProducts} />);

    const button = screen.getByRole("button", { name: "Add To Cart" });
    userEvent.click(button);
    expect(updateProducts).toHaveBeenCalledTimes(1);
    expect(fakeItem.increment).toHaveBeenCalledTimes(1);
  });
});
