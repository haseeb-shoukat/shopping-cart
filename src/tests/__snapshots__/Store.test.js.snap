// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store component Renders successfully 1`] = `
<div>
  <div
    class="products"
  >
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/cap.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Matt Black Cap, Unbranded, Unisex
          </div>
          <div
            class="product-price"
          >
            $
            19.50
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/jacket.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Nike Orange-White Jacket
          </div>
          <div
            class="product-price"
          >
            $
            155.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/tshirt.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Plain White T-Shirt for Men
          </div>
          <div
            class="product-price"
          >
            $
            60.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/dress.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Red Butterfly Dress + Black Belt for Women
          </div>
          <div
            class="product-price"
          >
            $
            245.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/glasses.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Rayban 2020 Limited Edition Shades
          </div>
          <div
            class="product-price"
          >
            $
            250.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/jeans.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Stylish Denim Jeans for Women, Tight Fit
          </div>
          <div
            class="product-price"
          >
            $
            30.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/polo.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Pink Polo Shirt, Slim Waist for Women
          </div>
          <div
            class="product-price"
          >
            $
            20.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/shirt.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Plain Black Shirt, Fitted, Men
          </div>
          <div
            class="product-price"
          >
            $
            40.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/shoes.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Stylish High Heels, Full Black
          </div>
          <div
            class="product-price"
          >
            $
            250.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt=""
        class="card-img"
        src="./images/clothing/suit.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-info"
        >
          <div
            class="product-name"
          >
            Class Release, Two Piece Black Suit
          </div>
          <div
            class="product-price"
          >
            $
            400.00
          </div>
        </div>
        <button
          class="cart-btn"
        >
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
